UI/js/controllers/controllers.js

```diff
@@ -125,6 +125,22 @@ dashboardApp.controller('scheduleController', ['Schedule', '$scope', '$rootScope
   Schedule.getSchedule('/todolms/' + $rootScope.user + '/canvas').then(function(data) {
     $scope.loadingSchedule = false;
     if(data.status ===200){
+      $.each(data.data.Result, function() {
+        if(this.contextLMS === 'canvas'){
+          this.context = null;
+        }
+      });
+      var canvasArray = shareCanvas.getCanvasArray();
+      $.each(data.data.Result, function() {
```

Comment:
No. But the service should update the Schedule values when the Courses query finishes (before or after).